### ISSUE TITLE
HttpClient is disposed every time

### ISSUE THREAD

**Issue by rokondra** [AzureAD/azure-activedirectory-library-for-dotnet#1459]
**ADAL 3.19.8**

**Which platform has the issue?**
net45

**What authentication flow has the issue?**
* Desktop / Mobile
    * [ ] Interactive
    * [ ] Integrated Windows Auth
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web App
    * [ ] Authorization code
    * [ ] OBO
* Web API
    * [x] OBO, I guess. The one that uses AAD applications and certificates

**Repro**
Inside Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Http.HttpClientWrapper
```csharp
	public async Task<IHttpWebResponse> GetResponseAsync()
	{
		using (HttpClient client = new HttpClient(HttpMessageHandlerFactory.GetMessageHandler(this.UseDefaultCredentials)))
		{

```

**Expected behavior**
It is more like a question, if this behavior is expected or not.
I would expect re-usable single HttpClient instance, so that it won't re-open/re-negotiate everything required everytime you request a new token (specifically in multi-tenant scnearios)

**Actual behavior**
Whenever the HttpClient is disposed, connection is closed. A new connection is required and all the overhead associated with that.

**Possible Solution**
Re-use the HttpClient instance.

**Additional context/ Logs / Screenshots**
Add any other context about the problem here, such as logs and screebshots. Logging is described at https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/wiki/Logging-in-ADAL.Net

**Comment by MatthewSteeples** [comment]
Internet seems to agree with you - https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/

**Comment by jmprieur** [comment]
This was fixed in ADAL 4.x (which is source compatible with 3.19.8)
Also MSAL.NET does even better.

**Comment by rokondra** [comment]
> This was fixed in ADAL 4.x (which is source compatible with 3.19.8)
> Also MSAL.NET does even better.

We took the latest available ADAL 4.4.2.0 and the issue seems still there, are we missing something?
```
var authenticationContext = new AuthenticationContext("https://login.windows.net/tenantid");
var certBase64 = "";

X509Certificate2 cert = new X509Certificate2(Convert.FromBase64String(certBase64), "<password>");
ClientAssertionCertificate clientAssertion = new ClientAssertionCertificate("appId", cert);

AuthenticationResult userToken = await authenticationContext.AcquireTokenAsync("audience", clientAssertion);
```
